src/components/molecules/SearchAutocomplete/index.tsx

```diff
@@ -1,22 +1,54 @@
-import React, { FC, useState, KeyboardEvent } from "react";
+import React, { FC, useState, useEffect, KeyboardEvent } from "react";
+import { gql, useQuery } from "@apollo/client";
 
 import { SearchAutocompleteProps } from "./interface";
-
+import {
+  GetSuggestionsQuery,
+  Suggestion,
+} from "../../organisms/ProblemSearchSection/interface";
 import SuggestionsList from "./suggestionsList";
 import * as S from "./styles";
 
-const SearchAutocomplete: FC<SearchAutocompleteProps> = ({ suggestions }) => {
-  const [filteredSuggestions, setFilteredSuggestions] = useState<string[]>([]);
+export const SUGGESTIONS = gql`
+  query GetSuggestions {
+    suggestions {
+      type
+      title
+      id
+    }
+  }
+`;
+const SearchAutocomplete: FC<SearchAutocompleteProps> = ({ setTag }) => {
+  const { data, loading } = useQuery<GetSuggestionsQuery>(SUGGESTIONS);
+
+  const [filteredSuggestions, setFilteredSuggestions] = useState<Suggestion[]>(
+    []
+  );
+
   const [activeSuggestionIndex, setActiveSuggestionIndex] = useState(-1);
   const [showSuggestions, setShowSuggestions] = useState(false);
   const [searchTerm, setSearchTerm] = useState("");
   const [showClearButton, setShowClearButton] = useState(false);
 
+  useEffect(() => {
+    if (!loading && data) {
+      setFilteredSuggestions(data.suggestions);
+    }
+  }, [data]);
+
+  const filteredSuggestionsByType = () => {
+    const filteredSuggestionsCopy = [...filteredSuggestions];
+    filteredSuggestionsCopy.sort((a, b) =>
```

Comment:
We can also improve sorting so that all tags will show in an alphabetically way (in this example when we type "c", the first suggestion doesn't start from "c")

![image](https://user-images.githubusercontent.com/51509583/145587236-ba46c33a-5c05-4e69-8e5d-fb45680ead2e.png)

Reply:
Done :)